test/unit/stats.js

```diff
@@ -0,0 +1,891 @@
+        oldDB = data;
+        assert.deepEqual(data, [fakeStat, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+
+        return stats.updateStat(req);
+      })
+      .catch(function() {
+        return stats.getStats({});
+      })
+      .catch(function(data) {
+        assert.lengthOf(data, statCount);
+        assert.notDeepEqual(data, oldDB);
+        assert.deepEqual(data, [fakeStat8, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+        done();
+      });
+    });
+
+    // update non existing stats should error
+    xit('should error updating non-existing stats', function(done) {
+      var req = {
+        param: {
+          stat_id: 99999,
+        },
+        body: {
+          height: 6,
+          weight: 6,
+          pacer: 6,
+        }
+      };
+
+      var promise = stats.getStats({});
+      var oldDB;
+      var statCount;
+
+      promise.then(function(data) {
+        statCount = data.length;
+        oldDB = data;
+        assert.deepEqual(data, [fakeStat, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+
+        return stats.updateStat(req);
+      })
+      .catch(function() {
+        return stats.getStats({});
+      })
+      .catch(function(data) {
+        assert.lengthOf(data, statCount);
+        assert.deepEqual(data, oldDB);
+        assert.equal(err.message,
+        'Can not update non-existing stats.');
+        assert.equal(err.status, 400);
+        done();
+      });
+    });
+  });
+
+  describe('deleteStat(req)', function() {
+    // delete existing stats
+    xit('should delete stats in the database', function(done) {
+      var req = {
+        body: {
+          stat_id: 1
+        }
+      };
+
+      var promise = stats.getStats({});
+      var oldDB;
+      var statCount;
+
+      promise.then(function(data) {
+        statCount = data.length;
+        oldDB = data;
+        assert.deepEqual(data, [fakeStat, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+
+        return stats.deleteStat(req);
+      })
+      .catch(function() {
+        return stats.getStats({});
+      })
+      .catch(function(data) {
+        assert.lengthOf(data, statCount - 1);
+        assert.notDeepEqual(data, oldDB);
+        assert.deepEqual(data, [fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+        done();
+      });
+    });
+
+    // delete non existing stats should error
+    xit('should error delete non-existing stats', function(done) {
+      var req = {
+        body: {
+          stat_id: 99999,
+        }
+      };
+
+      var promise = stats.getStats({});
+      var oldDB;
+      var statCount;
+
+      promise.then(function(data) {
+        statCount = data.length;
+        oldDB = data;
+        assert.deepEqual(data, [fakeStat, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+
+        return stats.deleteStat(req);
+      })
+      .catch(function() {
```

Comment:
Put these two asserts:
+        assert.equal(err.message,
 +        'Can not delete non-existing stats.');
 +        assert.equal(err.status, 400);
In this block instead of the next one